examples/snippets/partitioned-dml/application.rb

```diff
@@ -25,6 +25,59 @@ def self.run
       count = Album.delete_all
       puts "Deleted #{count} albums"
     end
+    puts ""
+    puts "Deleting all singers in the database using normal Read-Write transaction with PDML fallback"
+    #
+    # This example demonstrates using `isolation: :fallback_to_pdml`.
+    #
+    # --- HOW IT WORKS ---
+    # 1. Initial Attempt: The transaction starts as a normal, atomic, read-write transaction.
+    #
+    # 2. The Trigger: If that transaction fails with a `TransactionMutationLimitExceededError`,
+    #    the adapter automatically catches the error.
+    #
+    # 3. The Fallback: The adapter then retries the ENTIRE code block in a new,
+    #    non-atomic Partitioned DML (PDML) transaction.
+    #
+    # --- WARNING: CRITICAL USAGE REQUIREMENTS ---
+    # This implementation retries the whole transaction block without checking its contents.
+    # The user of this feature is responsible for ensuring the following:
+    #
+    # 1. SINGLE DML STATEMENT: The block SHOULD contain only ONE DML statement.
+    #    If it contains more, the PDML retry will fail with a low-level `seqno` error.
+    #
+    # 2. IDEMPOTENCY: The block MUST be "idempotent" (safe to run multiple times),
+    #    as the code may be executed more than once.
+    #
+    # 3. NON-ATOMIC: The retried PDML transaction is NOT atomic. Do not use this
+    #    for multi-step operations that must all succeed or fail together.
+    #
+    Singer.transaction isolation: :fallback_to_pdml do
+      count = Singer.delete_all
+      puts "Deleted #{count} singers"
+    end
+    Singer.transaction isolation: :fallback_to_pdml do
+      begin
+        singers_to_create = (1..10).map { |i| { first_name: "Test", last_name: "Singer #{i}" } }
+        Singer.create singers_to_create
+        puts "  #{Singer.count} singers now in database."
+
+        puts "\n  Running a large delete operation with 'isolation: :fallback_to_pdml'..."
+        puts "  NOTE: A real operation of this type on millions of rows could fail with a mutation limit error."
+        puts "  The adapter would catch this error and automatically retry with a PDML transaction."
+
+        Singer.transaction isolation: :fallback_to_pdml do
+          Singer.where("last_name LIKE 'Singer %'").delete_all
+        end
+
+        puts "\n  SUCCESS: The transaction completed successfully thanks to the PDML fallback."
+        puts "  Remaining singers: #{Singer.count}"
+      rescue StandardError => e
+        puts "\n  FAILED: The transaction unexpectedly failed with error: #{e.message}"
+      ensure
+        Singer.delete_all
+      end
```

Comment:
nit: I think that we should remove this entire block as well. The first transaction that deletes all singers is enough to show how this feature works.